atomic_reactor/plugins/pre_fetch_sources.py

```diff
@@ -252,6 +323,26 @@ def get_remote_urls(self):
 
         return remote_sources_urls, remote_sources_map
 
+    def get_kojifile_source_urls(self):
+        """Fetch kojifile source urls from all builds
+
+        :return: list, dicts with URL pointing to kojifile source files
+        """
+        kojifile_sources = []
+
+        kojifile_source = self._get_kojifile_source_urls_helper(self.koji_build)
+        kojifile_sources.extend(kojifile_source)
+
+        koji_build = self.koji_build
+
+        while 'parent_build_id' in koji_build['extra']['image']:
+            koji_build = self.session.getBuild(koji_build['extra']['image']['parent_build_id'],
+                                               strict=True)
+            kojifile_source = self._get_kojifile_source_urls_helper(self.koji_build)
```

Comment:
this should use koji_build not self.koji_build, as that is infinite loop because that doesn't ever change

Reply:
fixed it.